prow/scripts/lib/gardener/azure.sh

```diff
@@ -0,0 +1,330 @@
+#!/usr/bin/env bash
+
+# shellcheck source=prow/scripts/lib/log.sh
+source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/log.sh"
+# shellcheck source=prow/scripts/lib/kyma.sh
+source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/kyma.sh"
+# shellcheck source=prow/scripts/lib/utils.sh
+source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/utils.sh"
+# shellcheck source=prow/scripts/lib/docker.sh
+source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/docker.sh"
+# shellcheck disable=SC1090
+source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/cluster-integration/helpers/fluent-bit-stackdriver-logging.sh"
+
+#!Put cleanup code in this function! Function is executed at exit from the script and on interuption.
+gardener::cleanup() {
+    #!!! Must be at the beginning of this function !!!
+    EXIT_STATUS=$?
+    if [ "${ERROR_LOGGING_GUARD}" = "true" ]; then
+        log::error "AN ERROR OCCURED! Take a look at preceding log entries."
+    fi
+
+    #Turn off exit-on-error so that next step is executed even if previous one fails.
+    log::info "Cleanup"
+    set +e
+
+    log::info "Describe nodes"
+    kubectl describe nodes
+    kubectl top nodes
+    kubectl top pods --all-namespaces
+
+    # collect logs from failed tests before deprovisioning
+    kyma::run_test_log_collector "kyma-integration-gardener-azure"
+
+    if [[ -n "${SUITE_NAME}" ]]; then
+        kyma::test_summary
+        SUITE_EXIT_STATUS=$?
+        if [[ ${EXIT_STATUS} -eq 0 ]]; then
+            EXIT_STATUS=$SUITE_EXIT_STATUS
+        fi
+    fi
```

Comment:
At this point I'm wondering if the getting test summary is really the "cleanup" procedure. This just doesn't look good...

Reply:
test summary was probably moved here to get executed even when script encounters an error, maybe it could be separated and get only executed when script does not fail?

Reply:
Maybe, but we should see wether the CLI is throwing an error when the test fail. If so then maybe we should think about trapping another function that collects the test summary, then does a cleanup since cleanup shouldn't collect test results.

Reply:
These lines could be moved to `gardener::test_kyma` in slightly modified form:

```bash
gardener::test_kyma() {
    log::info "Running Kyma tests"

    readonly SUITE_NAME="testsuite-all-$(date '+%Y-%m-%d-%H-%M')"
    readonly CONCURRENCY=5
    set +e
    (
    set -x
    kyma test run \
        --name "${SUITE_NAME}" \
        --concurrency "${CONCURRENCY}" \
        --max-retries 1 \
        --timeout 120m \
        --watch \
        --non-interactive
    )

    # collect logs from failed tests before deprovisioning
    kyma::run_test_log_collector "kyma-integration-gardener-azure"

    if [[ -n "${SUITE_NAME}" ]]; then
        kyma::test_summary
        SUITE_EXIT_STATUS=$?
    fi

    set -e
    
    if [[ ${SUITE_EXIT_STATUS} -ne 0 ]]; then
        return $SUITE_EXIT_STATUS
    else
        log::success "Tests completed"
    fi
}
```

Reply:
That's the idea. however we can simplify the function:
```sh
gardener::test_kyma() {
    log::info "Running Kyma tests"

    readonly SUITE_NAME="testsuite-all-$(date '+%Y-%m-%d-%H-%M')"
    readonly CONCURRENCY=5
    set +e
    (
    set -x
    kyma test run \
        --name "${SUITE_NAME}" \
        --concurrency "${CONCURRENCY}" \
        --max-retries 1 \
        --timeout 120m \
        --watch \
        --non-interactive
    )

    # collect logs from failed tests before deprovisioning
    kyma::run_test_log_collector "kyma-integration-gardener-azure"
    if ! kyma::test_summary; then
      log::error "Tests have failed"
      return 1
    fi
    set -e
    log::success "Tests completed"
}
``` if `kyma::test_summary` returns an exit code (which should...) then this should be enough to check if the command had failed with non-0 exit code

Reply:
I'll update lib/gardener scripts, `if` will require `set -e` to work as expected:
```bash
    if ! kyma::test_summary; then
        log::error "Tests have failed"
        set -e
        return 1
    fi
```

Reply:
test_summary and _run_test_log_collector were moved to gardener::test_kyma